src/Sentry.AspNetCore/SentryMiddleware.cs

```diff
@@ -91,12 +95,14 @@ public async Task InvokeAsync(HttpContext context, RequestDelegate next)
                 context.Response.OnCompleted(() => hub.FlushAsync(_options.FlushTimeout));
             }
 
-            if (!_options.IsTracingEnabled)
-            {
-                var traceHeader = context.TryGetSentryTraceHeader(_options);
-                var baggageHeader = context.TryGetBaggageHeader(_options);
-                hub.ContinueTrace(traceHeader, baggageHeader);
-            }
+            var traceHeader = context.TryGetSentryTraceHeader(_options);
+            var baggageHeader = context.TryGetBaggageHeader(_options);
+            var transactionContext = hub.ContinueTrace(traceHeader, baggageHeader);
+
+            // Adding the headers and thw TransactionContext to the context to be picked up by the Sentry tracing middleware
+            context.Items.Add(TraceHeaderItemKey, traceHeader);
+            context.Items.Add(BaggageHeaderItemKey, baggageHeader);
+            context.Items.Add(TransactionContextItemKey, transactionContext);
```

Comment:
This is kind of an interesting (ab)use of `HttpContext.Items`... we're sort of using it a bit like a Post Office Box here to send stuff between our various Middleware. This will probably work without any issues (as long as the Key's you're picking never conflict with anything), but I don't think that's [what it was designed for](https://learn.microsoft.com/en-us/dotnet/api/system.web.httpcontext.items?view=netframework-4.8.1). 

Another (I think cleaner) option would be [ConditionalWeakTable](https://learn.microsoft.com/dotnet/api/system.runtime.compilerservices.conditionalweaktable-2). You could store the the `HttpContext` as the Key in that table and maybe a tupple holding the three values that you want to pass as the Value. 

You can see an [example of this in our codebase here](https://github.com/getsentry/sentry-dotnet/blob/dc0e0223aa64555966ff5c9ab194a535dc737b72/src/Sentry/Internal/Hub.cs#L21).

Reply:
Based on [this article](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/app-state?view=aspnetcore-7.0#httpcontextitems) the `items` seem to be the mechanism with which to pass data between middlewares.
I guess we could use objects to use as keys to be extra sure not to have any conflicting keys?